src/rest_vol_file.c

```diff
@@ -544,9 +551,44 @@ RV_file_get(void *obj, H5VL_file_get_args_t *args, hid_t dxpl_id, void **req)
 
         /* H5Fget_obj_ids */
         case H5VL_FILE_GET_OBJ_IDS:
-            FUNC_GOTO_ERROR(H5E_FILE, H5E_UNSUPPORTED, FAIL, "H5Fget_obj_ids is unsupported");
-            break;
 
+            if (args->args.get_obj_ids.max_objs <= 0)
+                FUNC_GOTO_ERROR(H5E_FILE, H5E_BADVALUE, FAIL, "invalid max object parameter");
+
+            if (args->args.get_obj_ids.oid_list == NULL)
+                FUNC_GOTO_ERROR(H5E_FILE, H5E_BADVALUE, FAIL, "given object id list buffer is NULL");
+
+            if (args->args.get_obj_ids.count == NULL)
+                FUNC_GOTO_ERROR(H5E_FILE, H5E_BADVALUE, FAIL, "given object id count buffer is NULL");
+
+            *args->args.get_obj_ids.count = 0;
+            unsigned int requested_types  = args->args.get_obj_ids.types;
+
+            object_id_list id_list;
+            id_list.obj_id_list = args->args.get_obj_ids.oid_list;
+            id_list.obj_count   = 0;
+
+            if (args->args.get_obj_ids.types & H5F_OBJ_FILE)
+                if (H5Iiterate(H5I_FILE, RV_iterate_copy_hid_cb, &id_list) < 0)
+                    FUNC_GOTO_ERROR(H5E_FILE, H5E_OBJECTITERERROR, FAIL, "can't iterate over file ids");
+
+            if (args->args.get_obj_ids.types & H5F_OBJ_GROUP)
+                if (H5Iiterate(H5I_GROUP, RV_iterate_copy_hid_cb, &id_list) < 0)
+                    FUNC_GOTO_ERROR(H5E_FILE, H5E_OBJECTITERERROR, FAIL, "can't iterate over group ids");
+
+            if (args->args.get_obj_ids.types & H5F_OBJ_DATATYPE)
+                if (H5Iiterate(H5I_DATATYPE, RV_iterate_copy_hid_cb, &id_list) < 0)
+                    FUNC_GOTO_ERROR(H5E_FILE, H5E_OBJECTITERERROR, FAIL, "can't iterate over datatype ids");
+
+            if (args->args.get_obj_ids.types & H5F_OBJ_DATASET)
+                if (H5Iiterate(H5I_DATASET, RV_iterate_copy_hid_cb, &id_list) < 0)
+                    FUNC_GOTO_ERROR(H5E_FILE, H5E_OBJECTITERERROR, FAIL, "can't iterate over dataset ids");
+
+            if (args->args.get_obj_ids.types & H5F_OBJ_ATTR)
+                if (H5Iiterate(H5I_ATTR, RV_iterate_copy_hid_cb, &id_list) < 0)
+                    FUNC_GOTO_ERROR(H5E_FILE, H5E_OBJECTITERERROR, FAIL, "can't iterate over attribute ids");
```

Comment:
After all the H5I calls have been made, the value of `id_list.obj_count` needs to be assigned back to `args->args.get_obj_ids.count` so that the value is properly returned.